### PR TITLE
[v16] Adding extra pod labels to post-upgrade and post-delete hook job pods

### PR DESCRIPTION
**Pull request by mvbrock**
Backport #46029 to branch/v16

changelog: Ensure that additional pod labels are carried over to post-upgrade and post-delete hook job pods when using the teleport-kube-agent Helm chart.

### CODE REVIEW

**Review by tigrato — APPROVED** [review]

**Review by hugoShaka — CHANGES_REQUESTED** [review]
Please don't merge those backports PR yet, this will break CI

**Comment by mvbrock** [comment]
> Please don't merge those backports PR yet, this will break CI

@hugoShaka can you give a little more detail? Should I fix something before merging?

**Comment by hugoShaka** [comment]
Sorry, I forgot to ping you, the fix passed, You can now rebase on branch/v16. The CI now enforces the chart documentation to be in sync with the chart values, it won't let the PR break the tests anymore.

After rebasing you might face a CI error asking to re-generate the docs, it will contain the command to do so.

**Review by hugoShaka — APPROVED** [review]

**Pull request merged**
